### ISSUE TITLE
Can't look up components inside of PolymerTemplate

### ISSUE THREAD

**Issue by mvysny** [mvysny/karibu-testing#1]
Yes. This is because of how Vaadin 10 currently works. The problem in here is that any DOM manipulation conducted by browser will not make its way into Vaadin server (even with a full Vaadin 10 server running, so this is not just an unfortunate effect of browserless testing). The implications are many.

The original issue was that `PolymerTemplate.getChildren()` streams returns no components even when there are components nested inside of the PolymerTemplate. This has been explained in https://github.com/vaadin/flow/issues/3642 - PolymerTemplate nests its contents inside of a Shadow DOM and Shadow DOM elements do not count as children.

Then I thought I could simply enumerate the contents of the template as suggested in this comment: https://github.com/vaadin/flow/issues/3642#issuecomment-370695179
And it worked, however the Buttons returned had an empty text!

And then the following bug came into play: https://github.com/vaadin/vaadin-button-flow/issues/43#issuecomment-370706462

In short - PolymerTemplate is parsed client-side and thus the DOM is not transferred server-side. On server-side, only the components mapped with `@Id` are constructed server-side; and those are nothing more but empty shells. So, for example the Button nests caption in a `<span>` element; however since DOM is not transferred the server-side Button thinks there is no span and therefore no caption and returns an empty text. Similarly all layouts (`VerticalLayout`) will have no children since it detects its children by enumerating its element's DOM children, but there is no DOM.

**Comment by mvysny** [comment]
So, to sum it up:

* `PolymerTemplate.getChildren()` returns an empty list and it's going to stay like that.
* even if I used the `VirtualChildrenList` workaround, I would not able to enumerate all components inside of the PolymerTemplate since there will be no server-side counterpart constructed for any component without an Id; so no Button for `<vaadin-button>` without id;
* Even if I would be able to enumerate all of the components in a `PolymerTemplate`, it would be useless since you wouldn't be able to find a Button based on the caption since there is no DOM and the `Button.text` will be empty.

The only way is to access `@Id`-annotated fields in the `PolymerTemplate` directly, for example:

```kotlin
class ReviewsList : PolymerTemplate<TemplateModel>() {
    @Id("search")
    internal lateinit var search: TextField
    // ...
}
test("create review") {
   val list = ReviewsList()
   list.addReview._click()
}
```

**Comment by mvysny** [comment]
Can't fix - this is unfortunately how Vaadin 10 works and there is nothing I can do about it.

**Comment by mvysny** [comment]
Example in Java:

```java
public class ReviewsList extends PolymerTemplate<TemplateModel> {
    @Id("search")
    TextField search;
    // ...
}

@Test
public void createReviewTest() {
   ReviewsList list = _get(ReviewsList.class);
   _click(list.addReview);
}
```
